torch/utils/data/sampler.py

```diff
@@ -89,6 +89,7 @@ def __init__(self, data_source: Sized, replacement: bool = False,
         self.replacement = replacement
         self._num_samples = num_samples
         self.generator = generator
+        self.gen: Optional[torch.Generator] = None
```

Comment:
Maybe, `self.gen` can be a private attribute (no worries about later changing and BC etc) ?

Reply:
Makes sense to me.
BTW, adding `gen` to self may provide a way to resume in the middle of epoch by serializing/deserializing the state of `gen`, since the `gen` is attached to the sampler instance now.